### DISCUSSION TITLE
skyline not working

### DISCUSSION THREAD

**Discussion by khopland** [community/community#64942] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Bug

### Body

looks like [skyline](https://skyline.github.com/) is not working anymore, is this a bug in the code or is it something that has been deprecated?

**Comment by paras-017** [comment]
when i type my username its shows : We did not find any @paras-017 on GitHub, try it again

**Reply by davidteren** [reply]
I get the same error.

**Reply by balnarendrasapa** [reply]
I am getting same error and I have also tried other usernames too. but it seems there is something wrong on their side.

**Reply by AndusDEV** [reply]
Same here

**Reply by GeoJunkie** [reply]
Same. I thought it was my mixed-case username, but apparently it's just not working.

**Reply by PlainJi** [reply]
+1
I am using this API to show my contribution wall on a small screen.
Not working, sad...
Any other API can be used?

**Reply by z1097955019** [reply]
so, is there anything instead

**Reply by emilkrebs** [reply]
Having the same issue. Any updates on this?

**Reply by resonanceee** [reply]
any updates?

**Reply by TariqSaiyad** [reply]
Same here, not working for me

**Reply by avikalpg** [reply]
same here. I am getting the same error: "We did not find any @avikalpg on GitHub, try it again."

**Reply by ashesfall** [reply]
same issue

**Reply by thedocruby** [reply]
Same here. Why has this not been addressed?

**Reply by ghaithtroudi** [reply]
same here

**Reply by Hao-yiwen** [reply]
same here.

**Reply by lucaspalomodevelop** [reply]
same here

**Reply by KBRR11** [reply]
same here

**Reply by neuralgitwork** [reply]
same here

**Reply by Dityaren** [reply]
same here

**Reply by yashksaini-coder** [reply]
same here also how come i saw a gif of github skyline although it gives the download in a STL file

**Reply by avikalpg** [reply]
> same here also how come i saw a gif of github skyline although it gives the download in a STL file
> 

What you see on GitHub Skyline is not a GIF. It is a 3D animation that you can also interact with. The STL file represents the 3D object, which is your skyline.

(You can refer to my open source clone of this project to understand that in more detail: https://github.com/avikalpg/skyline)

**Reply by ebndev** [reply]
Hi all! In case you aren't aware, we've turned the skyline microsite off. If you have any questions about this, please respond to [this thread](https://github.com/orgs/community/discussions/129740#discussioncomment-9873873) in Discussions.

**Comment by nicolaerario** [comment]
I bought the 3d printer just for the skyline 😩

**Reply by PlainJi** [reply]
Same idea

**Reply by Hanrryshani** [reply]
same

**Comment by jonathanpiette** [comment]
Same here, waiting for a fix :)

**Comment by ToasterUwU** [comment]
Same here. Just got my first 3D Printer, was so excited to make my first skyline

**Reply by enisn** [reply]
Exactly the same situation, I just got my first 3D printer and was so excited to print my own skyline but it's not working anymore...

**Comment by ahpxex** [comment]
seems like lots of people got this problem lol, so do i

**Comment by SoftwareWajih** [comment]
Github Skyline is not finding the username.

**Comment by Ahabibullah** [comment]
still not working unfort

**Comment by Synex93** [comment]
Is there still an alternative API for the problem

**Comment by mironovmeow** [comment]
Still not working .-.

---

I found the source code of the broken API: https://github.com/martinwoodward/json-contributions. Maybe it will be useful for someone.

**Reply by DRodrigues17** [reply]
did you could do a skyline by using this ? is tried but didn't have success

**Reply by avikalpg** [reply]
Yes, the repository doesn't seem to be doing anything much. When I try using their vercel app, it always only sends 0s in the response. Like so:
```json
{
    "username": "avikalpg",
    "year": "2021",
    "min": 0,
    "max": 0,
    "median": 0,
    "p80": 0,
    "p90": 0,
    "p99": 0,
    "contributions": []
}
```

**Comment by jyxjjj** [comment]
@github

**Comment by ZjzMisaka** [comment]
They didn't even announce that it had stopped running or was under maintenance

**Reply by jyxjjj** [reply]
I had submitted a ticket and they had already transferred to that team but now have nothing else happened.

**Reply by ZjzMisaka** [reply]
Any updates on this?

**Comment by avikalpg** [comment]
@Github talks so much about open source but didn't make this project open source. What a shame. We could at least have run it locally using the GitHub API. I was also excited to 3D print my skyline.

**Comment by onshisan** [comment]
What a shame that this fun little app has broken.

**Comment by ftrauernicht** [comment]
I always printed my Skyline at the end of the year and place them on my desk as a little trophy for all the work i have done that year :/

**Comment by supercodingninja** [comment]
My fellow coders, what if we all collaborated, together; and made something better than the Skyline: I always desired the whole history to show.  Who is with me?!  I can use a good project.  I'm trying to get hired.  Plus, I would like to make more IT friends.  Ping me 👨🏾‍🎓

**Reply by avikalpg** [reply]
Yes, I don't think it is very challenging. I also thought about this earlier. 
Just created [a repository for this](https://github.com/avikalpg/skyline). I think we can easily run this as a ReactJS application (the original application was in NuxtJS (VueJS), but I don't see the need for server-side work here. The entire application can run client-side. All it has to do is get the data from the GitHub API and render the 3D design. The rendering is going to happen on the browser anyway.)

Here is the link again: https://github.com/avikalpg/skyline

I'd love your participation and everyone's input.

**Reply by avikalpg** [reply]
If you are interested in contributing, please reply to this comment to get yourself added as a contributor in this repository.
I have created a couple of pull requests.

**Reply by ashesfall** [reply]
Sure, I might work on it. I don't have any frontend experience, but I could probably make progress on conversion from the API data to 3d model data.

**Reply by ToasterUwU** [reply]
I would love to help, but you can practically hunt me with JS of any kind, so sadly, I'm out.

**Reply by ftrauernicht** [reply]
I would love to see progress on this, unfortunatly my Personal time is very limited so i am out as well :/ glad that you are working on this!

**Reply by yourjhay** [reply]
I'll try this on my local :)

**Reply by avikalpg** [reply]
Hey everyone, the project is ready now. You can run it locally using ReactJS right now. I have used ThreeJS for the 3D rendering. 

[Link to the repository again](https://github.com/avikalpg/skyline)

**Reply by ftrauernicht** [reply]
I just pulled the repo and tried it for the first time, works like a charm! Thanks!
Now waiting for the .stl export :)

**Reply by avikalpg** [reply]
Yes, one of my friends, @balub, owns a 3D printer and is working on the pull request to add that support.

**Comment by moritztim** [comment]
![image](https://github.com/community/community/assets/90388353/cea049ae-3846-47a9-a07e-2bb16ee55437)
Yeah... pretty sure it's broken

**Comment by moritztim** [comment]
I can imagine their token just expired :skull:

**Reply by ftrauernicht** [reply]
I was thinking something similar, earlier this year the github public auth key had to be changed globally, this affected all of my seven CI systems. Maybe this is part of the problem? Did it worked after 2023-03-24?

https://github.blog/2023-03-23-we-updated-our-rsa-ssh-host-key/

Personally i only access this tool at the upcoming end of the year.

**Reply by moritztim** [reply]
@ftrauernicht 

> we replaced our RSA SSH host key used to secure Git operations for GitHub.

As the header suggests, this regards SSH host keys for git, not anything to do with the API. What I'm suggesting is that they set their token to expire after a while, which is a reasonable thing to do and then just neglected to create a new one after it expired.

**Comment by onshisan** [comment]
> Did it worked after 2023-03-24?
> 
> https://github.blog/2023-03-23-we-updated-our-rsa-ssh-host-key/
> 
> Personally i only access this tool at the upcoming end of the year

I've been wondering this myself. Seems like a viable lead.

**Comment by ToasterUwU** [comment]
I made a Ticket, and apparently, they are taking care of it now. I gave them a link to this discussion so they know where to find us.

![image](https://github.com/community/community/assets/43654377/e9069f2d-c071-4264-8499-9357faf15301)

**Reply by ZjzMisaka** [reply]
"That doesn't mean we've forgotten about it"
pretty nice

**Comment by cr2007** [comment]
Just got this update 5 minutes ago

[GitHub Skyline](https://skyline.github.com/) is officially back in business 🥳 

![image](https://github.com/community/community/assets/73425927/31e98809-de83-4581-87d3-9054ca10f06a)

![image](https://github.com/community/community/assets/73425927/4b7fca8c-2a72-4526-99b0-c4654d65279a)

**Reply by DRodrigues17** [reply]
Now i Will print and save the file in my Google drive, just for precaution.

**Reply by ftrauernicht** [reply]
Thanks mate!

I would still love to see further progress on the self hosted project, just in case the offical one will be unavailabe again :)
@avikalpg

**Reply by SoftwareWajih** [reply]
Amazing nice job @cr2007

**Reply by avikalpg** [reply]
Nice! Thanks for the encouragement @ftrauernicht.
And with the self-hosted version, we can maybe also add support for other providers like GitLab, Bitbucket and local repos (all color-coder).

**Comment by cr2007** [comment]
[laugh]         Ramaprasad, CSK reacted to your message:
________________________________

**Comment by jyxjjj** [comment]
Officially continuing getting bugs:
1. cannot add annotations.
2. cannot share via directly link with a year number or a username.

**Reply by velllu** [reply]
I cannot add annotations either, i am using firefox

**Comment by Toshiwoz** [comment]
The share link does not work:
- I can see my skyline, but when I try to share, the [link sends to a 404](https://skyline.github.com/toshiwoz/2023).

**Comment by ENTPRESTIGIOUS** [comment]
There are various issues with GitHub Skyline

1) The music no longer plays
2) I get various console errors
3) Whenever I try to drag around and view my profile skyline, my mouse gets "stuck" and i have to double click to negate the rotation
4) When I refresh the page, it says user does not exist

**Comment by bitfactory-douwe-devries** [comment]
I tried using it today, it seems to work except that it does not show any commits (the skyline is just flat).

Console gives a 404 on trying to get:
`https://skyline.github.com//bitfactory-douwe-devries/2023.png`

**Comment by HuakunShen** [comment]
I had made a clone of github skyline https://git-skyline.huakun.tech/

**Reply by yashksaini-coder** [reply]
Thank you for this. Really loved your skyline

**Reply by ebndev** [reply]
@HuakunShen this is awesome!

**Reply by jyxjjj** [reply]
Thanks but, it is not for print due to the gaps between the pillars. Can you add different model type to choose a printable or more clarify readable both? And it does not have a base like original skyline did.

**Reply by HuakunShen** [reply]
> Thanks but, it is not for print due to the gaps between the pillars. Can you add different model type to choose a printable or more clarify readable both? And it does not have a base like original skyline did.

@jyxjjj I added a `base` option for URL param. Adding `?&base=true` will add a thin base to make the model printable. See README for more info.

**Reply by jyxjjj** [reply]
> @jyxjjj I added a `base` option for URL param. Adding `?&base=true` will add a thin base to make the model printable. See README for more info.

Well done, but if the commit number is `0`, it still is a pillar instead of a flat ground.

**Reply by jyxjjj** [reply]
> Well done, but if the commit number is `0`, it still is a pillar instead of a flat ground.

![image](https://github.com/user-attachments/assets/dd2aa033-cce9-4555-ae72-0a00b76a5501)

Original like:
![image](https://private-user-images.githubusercontent.com/73425927/287338020-4b7fca8c-2a72-4526-99b0-c4654d65279a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjEyMDExMjEsIm5iZiI6MTcyMTIwMDgyMSwicGF0aCI6Ii83MzQyNTkyNy8yODczMzgwMjAtNGI3ZmNhOGMtMmE3Mi00NTI2LTk5YjAtYzQ2NTRkNjUyNzlhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzE3VDA3MjAyMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA3MWMyMWVjNmIxMzNiNDcxNTc2N2Q3NzAxMDM3OGQzODJlZjMzMGQ1OGEyYmZlZGQ0ZjkyMDI1OTRmYjg0OTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.TVIu63iIviuVMNDgCzKOFkAilRDPzlWuPTYtER_-DI0)

still the gaps between the pillars

**Reply by HuakunShen** [reply]
@jyxjjj Fixed

**Reply by jyxjjj** [reply]
> @jyxjjj Fixed

Thank you, that's good, although still not the same like original, but it is already very good. Now i  only need a little edit (via autodesk series and so on). Well done.

**Reply by HuakunShen** [reply]
I don't have time to make it look like the original. If more people are using it I will improve it.

**Reply by AdnanMuhib** [reply]
Good Job @HuakunShen 👍 .

Is there a way to make the embed URL work in the Readme?

**Reply by HuakunShen** [reply]
I don’t think github markdown supports iframe.

**Reply by AdnanMuhib** [reply]
Github Markdown supports stl (ascii).
https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-diagrams#creating-stl-3d-models
I am experimenting with conversion of binary stl to ascii stl and test in markdown.
I will share the results if it works.

**Comment by ebndev** [comment, marked as answer]
Hi all! Excited to share that we’ve [released gh-skyline](https://github.com/github/gh-skyline), a GitHub CLI extension that allows you to render your contribution graph as an STL file for 3D printing!

If you used our former GitHub Skyline site, then this concept may be familiar to you! But we’ve sprinkled a few additional features into the CLI as well. Read all about it in [this discussion](https://github.com/orgs/community/discussions/146467) and please share any comments or questions directly in that post. 😄

**Reply by cr2007** [reply]
Merry Christmas indeed! 🎄 
Hope you and the `gh-skyline` team have a holly jolly one! 🫂
